paasta_tools/cli/schemas/kubernetes_schema.json

```diff
@@ -190,7 +190,103 @@
+                            ]
+                        },
+                        "decision_policy": {
+                            "type": "string"
+                        },
+                        "setpoint": {
+                            "type": "number"
+                        },
+                        "forecast_policy": {
+                            "enum": [
+                                "moving_average",
+                                "current"
+                            ]
+                        },
+                        "offset": {
+                            "type": "number"
+                        },
+                        "moving_average_window_seconds": {
+                            "type": "integer"
+                        },
+                        "use_prometheus": {
+                            "type": "boolean"
+                        },
+                        "use_resource_metrics": {
+                            "type": "boolean"
+                        },
+                        "uwsgi_stats_port": {
+                            "type": "integer"
+                        },
+                        "scaledown_policies": {
+                            "type": "object"
+                        },
+                        "good_enough_window": {
+                            "type": "array",
+                            "items": {
+                                "type": "number"
+                            },
+                            "minItems": 2,
+                            "maxItems": 2
+                        },
+                        "prometheus_adapter_config": {
+                            "type": "object",
+                            "properties": {
+                                "seriesQuery": {
+                                    "type": "string"
+                                },
+                                "metricsQuery": {
+                                    "type": "string"
+                                }
+                            },
+                            "required": [
+                                "metricsQuery"
+                            ]
+                        }
+                    },
+                    "allOf": [
+                        {
+                            "oneOf": [
+                                {
+                                    "properties": {
+                                        "metrics_provider": {
+                                            "enum": [
+                                                "uwsgi",
+                                                "mesos_cpu",
+                                                "cpu",
+                                                "piscina",
+                                                "http",
+                                                "if metrics_provider is arbitrary_prometheus, the prometheus_adapter_config parameter is required"
```

Comment:
This is here because without it, when you specify `metrics_provider: arbitrary_prometheus` but forget `prometheus_adapter_config`, you get
```
  Validation Message: 'arbitrary_prometheus' is not one of ['uwsgi', 'mesos_cpu', 'cpu', 'piscina', 'http']
```
I couldn't figure out any combination of `oneOf`/`anyOf`/`if` and `then` which would cause it to spit out the more helpful
```  Validation Message: 'prometheus_adapter_config' is a required property```


With this hack, you at least get
```
  Validation Message: 'arbitrary_prometheus' is not one of ['uwsgi', 'mesos_cpu', 'cpu', 'piscina', 'http', 'if metrics_provider is arbitrary_prometheus, the prometheus_adapter_config parameter is required']
```

Reply:
does jsonschema yell at us if we have comments in this file? would be nice to have this comment not just live in github :p

from the lack of trailing commas in this file, i'm gonna assume that jsonschema is pretty strict with the input file tho :(

Reply:
looks like we just use `json.loads`, which doesn't support comments, but jsonschema version 7 introduces a `$comment` field. http://json-schema.org/understanding-json-schema/reference/generic.html#comments (I'm not sure if we're using version 7 or just ignoring any unknown fields in the schema, but it seems to work.)